Afazir/benthic

afazir opened the pull request.

afazir committed:
Updating shapedetector to be more accurate

afazir committed:
Switch to Otsu Threshholding

afazir committed:
Updating code for benthic

afazir committed:
Adding output to benthic

afazir committed:
Adding output screen

afazir committed:
Final Version of Benthic Species

jcassarly commented on shapedetector.py:
```diff
@@ -1,5 +1,6 @@
+import imutils
+import numpy as np
 
 class ShapeDetector:
```
what does this class do? you should provide a description of the class with comments

also give description to all functions too

jcassarly commented on shapedetector.py:
```diff
@@ -20,7 +21,7 @@ def detect(self, c):
             print(ar)
             area = cv2.contourArea(c)
-            if area/float(w*h) <= 0.4 or ar <= 0.8 or ar >= 1/0.8:
+            if area/float(w*h) <= 0.4 or ar <= 0.8 or ar >= 1/0.75:
```
please dont use magic numbers.  pull these out as constants and give them names

jcassarly commented on shapedetector.py:
```diff
@@ -1,5 +1,6 @@
+import numpy as np
 
 class ShapeDetector:
     def __init__(self):
```
if youre just passing, you dont need this function, it will automatically be there.  you should only defined __init__ if you need to change how the constructor for the object works

jcassarly commented on shapedetector.py:
```diff
@@ -39,20 +40,22 @@ def add_shape(shape, d):
 
 def detect_shapes():
```
what is this function doing?

also add some whitespace and inline comments

jcassarly commented on shapedetector.py:
```diff
@@ -39,20 +40,22 @@ def add_shape(shape, d):
     gray = cv2.cvtColor(resized, cv2.COLOR_BGR2GRAY)
     blurred = cv2.GaussianBlur(gray, (5, 5), 0)
 
     # Here we use an adaptive threshold on the image, since we expect the lighting to be non-uniform.
```
```suggestion
    # Use an adaptive threshold on the image, since the lighting is expected to be non-uniform.
```

jcassarly commented on shapedetector.py:
```diff
@@ -39,20 +40,22 @@ def add_shape(shape, d):
     # Here we use an adaptive threshold on the image, since we expect the lighting to be non-uniform.
-    thresh = cv2.adaptiveThreshold(blurred, 255, cv2.ADAPTIVE_THRESH_GAUSSIAN_C, cv2.THRESH_BINARY, 5, 0)
-    cv2.imshow("thresh", thresh)
+    ret, otsu = cv2.threshold(blurred, 0, 255, cv2.THRESH_BINARY+cv2.THRESH_OTSU)
```
what is ret and otsu?

jcassarly commented on shapedetector.py:
```diff
@@ -66,7 +69,7 @@ def detect_shapes():
 
         # ignore contours that are too small to be species
-        if c.shape[0] > 2 and area/(resized.shape[0]*resized.shape[1]) > 0.002:
+        if c.shape[0] > 2 and area/(resized.shape[0]*resized.shape[1]) > 0.002 and area/(resized.shape[0]*resized.shape[1]) < 0.25:
```
again, no magic numbers please.

also, you might want to put this on multiple lines:
```suggestion
        if c.shape[0] > 2 
           and area/(resized.shape[0]*resized.shape[1]) > 0.002 
           and area/(resized.shape[0]*resized.shape[1]) < 0.25:
```

jcassarly commented on shapedetector.py:
```diff
@@ -66,7 +69,7 @@ def detect_shapes():
         # ignore contours that are too small to be species
-        if c.shape[0] > 2 and area/(resized.shape[0]*resized.shape[1]) > 0.002:
+        if c.shape[0] > 2 and area/(resized.shape[0]*resized.shape[1]) > 0.002 and area/(resized.shape[0]*resized.shape[1]) < 0.25:
             cx = int((M["m10"] / M["m00"]) * ratio) if M['m00'] != 0 else 0
```
what are these mxx's?

jcassarly commented on shapedetector.py:
```diff
@@ -77,15 +80,24 @@ def detect_shapes():
-            # cv2.waitKey(0)
-
+	
+    #Draw shapes
```
draw what shapes, also put a space between the # in a comment and the text
```suggestion
    # Draw shapes
```

jcassarly commented on shapedetector.py:
```diff
@@ -77,15 +80,24 @@ def detect_shapes():
-
+	
+    #Draw shapes
+    line = cv2.rectangle(blank,(25,25), (190,45), (0,0,255),-1)
```
no magic numbers

jcassarly commented on shapedetector.py:
```diff
@@ -77,15 +80,24 @@ def detect_shapes():
-
 num_shapes = detect_shapes()
-print(num_shapes)
+print(num_shapes)
```
add a new line at the end of files
```suggestion
print(num_shapes)

```

jcassarly commented on shapedetector.py:
```diff
@@ -77,15 +80,24 @@ def detect_shapes():
-            # cv2.imshow("Image", image)
-            # cv2.waitKey(0)
-
+	
```
you should consider putting this part of the script that is after all the function generators inside 
```python
if __name__ == "__main__":
```

this avoids your script being run if someone were to import this file.  it should probably only be run if it is the fun run

jcassarly replied:
i'm a little confused why you put the if clause around just a few lines.  assuming im interpreting your code correctly, the `if __name__` ... should be on this line

jcassarly commented:
also are all of the images that youre uploading with this necessary? you arent using them in most of your code.

afazir committed:
Remove Magic Numbers

aranyakumar committed:
commented on code

aranyakumar committed:
Merge branch 'afazir/benthic' of https://github.com/CWRUbotixROV/rov-vision into afazir/benthic

afazir committed:
Add comments

aranyakumar committed:
created inline comments

afazir committed:
Added comments and removed magic numbers

afazir committed:
Created inline Comments
Merge branch 'afazir/benthic' of https://github.com/CWRUbotixROV/rov-vision into afazir/benthic

jcassarly commented on shapedetector.py:
```diff
@@ -2,54 +2,69 @@
+        # Shapes can only be square, triangle, line, or circle
         print(len(approx))
+        
+        AREARATIO = 0.4
```
If your variables have multiple words you should put underscores between the words:
`AREARATIO` should be `AREA_RATIO`

jcassarly commented on shapedetector.py:
```diff
@@ -66,7 +84,14 @@ def detect_shapes():
+        LOWERCONTOURTHRESH = 0.002
+
+        if c.shape[0] > 2 and\
+        area/(resized.shape[0]*resized.shape[1]) > LOWERCONTOURTHRESH and\
```
you should align the other parts of the if clause with the first one, ie this line should be

```suggestion
           area/(resized.shape[0]*resized.shape[1]) > LOWERCONTOURTHRESH and\
```

jcassarly commented on shapedetector.py:
```diff
@@ -66,7 +84,14 @@ def detect_shapes():
+        area/(resized.shape[0]*resized.shape[1]) > LOWERCONTOURTHRESH and\
+        area/(resized.shape[0]*resized.shape[1]) < UPPERCONTOURTHRESH:
+            # find centroid of contour
+            # if M['m00'] == 0 shape = line
```
why is this code commented out? should it still be here? if not you should remove it

aranyakumar committed:
crops image but does not detect shapes

aranyakumar committed:
final final benthic
